src/integrators/integrator_utils.jl

```diff
@@ -205,9 +221,6 @@ end
       copyat_or_push!(integrator.sol.u,integrator.saveiter,integrator.u[integrator.opts.save_idxs],Val{false})
     end
   end
-  if (!isnothing(integrator.W) && integrator.W.curt != integrator.t) || (!isnothing(integrator.P) && integrator.P.curt != integrator.t)
-    accept_step!(integrator,false)
```

Comment:
This seems to be a good fix such that the last noise values are not missing in the final solution object